src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -125,4 +106,41 @@ private void configureClient(DeviceConfiguration deviceConfiguration) {
         }
         this.iotDataPlaneClient = iotDataPlaneClientBuilder.build();
     }
+
+    /**
+     * Getter for IoT data plane client. This configures the client everytime the getter is used.
+     *
+     * @return iotDataPlaneClient
+     * @throws IoTDataPlaneClientCreationException exception during client configuration
+     */
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException"})
+    public IotDataPlaneClient getIotDataPlaneClient() throws IoTDataPlaneClientCreationException {
+        try {
+            // To ensure that the http client is configured with mTLS, wait for the crypto key provider service (pkcs11)
+            // to load. If the service is not loaded even after retrying, we throw an exception.
+            waitForCryptoKeyServiceProvider();
+        } catch (Exception e) {
+            throw new IoTDataPlaneClientCreationException(e);
+        }
+        configureClient();
```

Comment:
I'm curious how expensive this is. I _think_ we've already had some complaints about ShadowManager not being able to keep up (am I making stuff up?) and I'd be nervous about this worsening that.

Reply:
I can revert it to the [previous state](https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/main/src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java#L66-L72) where client is configured only in the constructor and when the `deviceConfiguration` changes. Here we can wait for the crypto key provider service to load.

Reply:
Re-using is always preferable as long as we're not opening up other holes. If we re-init every time then I'd want to do some perf testing to ensure it's not negatively impacting us.

Reply:
removed calls from constructor and subscribe as they block the main context thread.